web/web/api/v1/login.py

```diff
@@ -0,0 +1,122 @@
+from flask import request, jsonify, Blueprint
+from werkzeug.security import generate_password_hash
+from marshmallow import ValidationError
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm.exc import NoResultFound, MultipleResultsFound
+from .response_wrapper import ApiResponseWrapper
+from web.database import db
+from web.models.login import Login, LoginSchema
+
+login_api_bp = Blueprint('login_api_bp', __name__)
+
+
+@login_api_bp.route('/login/<int:login_id>', methods=['GET'])
+def show_login_info(login_id):
+    '''
+    Retrieves one login object
+    '''
+    arw = ApiResponseWrapper()
+    login_schema = LoginSchema(exclude=['password_hash'])
+
+    try:
+        login = Login.query.filter_by(login_id=login_id).one()
+
+    except (MultipleResultsFound, NoResultFound):
+        arw.add_errors('No result found or multiple results found')
+
+    if arw.has_errors():
+        return arw.to_json(None, 400)
+
+    results = login_schema.dump(login)
+
+    return arw.to_json(results)
+
+
+@login_api_bp.route('/login/<int:login_id>', methods=['PUT'])
+def modify_login(login_id):
+    '''
+    Updates one login object in database
+    '''
+
+    arw = ApiResponseWrapper()
+    login_schema = LoginSchema(exclude=[
+        'updated_at', 'created_at'
+    ])
+    modified_login = request.get_json()
+
+    try:
+        Login.query.filter_by(login_id=login_id).one()
+        modified_login = login_schema.load(modified_login, session=db.session)
+        db.session.commit()
+
+    except (MultipleResultsFound, NoResultFound):
+        arw.add_errors('No result found or multiple results found')
+
+    except ValidationError as ve:
+        arw.add_errors(ve.messages)
+
+    except IntegrityError:
+        arw.add_errors('Integrity error')
+
+    if arw.has_errors():
+        db.session.rollback()
+        return arw.to_json(None, 400)
+
+    results = login_schema.dump(modified_login)
+
+    return arw.to_json(results)
+
+
+@login_api_bp.route('/validate_login', methods=['POST'])
```

Comment:
I think overall this is going in the right direction, but some things seem odd to me. This validate operation should really be just `/login`. I assume any user that already has an account should be able to just `/login` which should hit this endpoint, is that correct? It feels semantically different than `validate_login`. Also, this login operation should return a token of some sort. Will that be a diff PR with the JWT auth?

Reply:
Yes I named it /validate_login (which is a /login - checks the username/password hash combo, and raises an error if it is incorrect, but can change the endpoint to /login). I named it /validate_login to separate it from the /create_login post route which is used to add a new login. Yes the following PR will be the auth tokens.